### PR TITLE
docs: add Cloud Architecture Guidelines

### PR DESCRIPTION
**Pull request by diazGeoff**
## What does this PR do?
- Added cloud architecture guidelines specifically for Cloud IAM and Secret Manager

## Why did you choose this solution?
- To enhance security and efficiency in cloud service configurations

## What is the dependency graph of the changes?
- README.md
- standards/cloud_defaults.md
- standards/architecture/cloud_iam.md
- standards/architecture/secret_manager.md
- standards/architecture/assets/iam_grant_button.png
- standards/architecture/assets/iam_grant_modal.png
- standards/architecture/assets/iam_role.png
- standards/architecture/assets/secret_add.png
- standards/architecture/assets/secret_env_variables.png

## How to test/verify?
- Review the updated guidelines in the standards/cloud_defaults.md
- Check added assets for IAM and Secret Manager steps

## Related ticket url:
-

### CODE REVIEW

**Comment by symphdevtools** [comment]
Hi! I'm generating PR title and description for you and will automatically apply it to your PR. I'll react :rocket: to this comment when done. Just sit back and chill!

When you have new changes to this PR and you want to regenerate the title and PR description, just comment `churva`!

Let me know if something is not working well, e.g.:
- Some parts of the title and description weren't preserved when generating title and description
- Disappearing screenshots or links
- Etc.

Thanks!

(RIP hatdog)

File: standards/cloud_defaults.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Cloud Architecture Defaults
+
+The following standards outline the recommended configurations and practices for cloud-based services to ensure optimal security, scalability, and efficiency in line with industry best practices.
```
Comment (JHNLWHD) [inline standards/cloud_defaults.md:3]: @diazGeoff let's add the diagram given by @rmarfil3 here so it serves as a visual guide for our cloud defaults

File: standards/architecture/secret_manager.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+4. Click **REFERENCE A SECRET**.
+5. Fill in the following details for your secret
+    ![Add Secret](./assets//secret_add.png)
+    a. Create a name for your secret key, this is also what you will reference in your code.
+    
+    b. Choose a secret from the secrets manager
+
+    c. Choose the version of the secret that you selected in step b.
+6. Click **DONE**.
+7. At the very bottom, click **DEPLOY**. This will deploy a new revision of your app with the environment variables loaded before your container is instantiated.
```
Comment (JHNLWHD) [inline standards/architecture/secret_manager.md]: Can we add a GIF also?

File: standards/architecture/cloud_iam.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- **Role**: A role is a collection of permissions. Permissions determine what operations are allowed on a resource. When you grant a role to a principal, you grant all the permissions that the role contains.
+
+## How to grant access to users and service accounts
+- Goto the IAM & Admin page in the Google Cloud Console of the specific project.
+- Click the grant access button to add new principal.
+    ![Grant Access](./assets/iam_grant_button.png)
+- Add your principal's email address and assign a role.
+    ![Add Principal](./assets/iam_grant_modal.png)
+- Click save. This would be the sample output.
+    ![Sample Output](./assets/iam_role.png)
```
Comment (JHNLWHD) [inline standards/architecture/cloud_iam.md]: Can we add a GIF also?

**Review by JHNLWHD — CHANGES_REQUESTED** [review]

**Review by JHNLWHD — APPROVED** [review]
LGTM 👍🏼

File: standards/architecture/assets/iam_02.gif
Comment (ramidem) [inline standards/architecture/assets/iam_02.gif:1]: is this public? ok lang makita email mo?

**Pull request merged**